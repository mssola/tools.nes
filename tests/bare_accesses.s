.segment "HEADER"
    .byte 'N', 'E', 'S', $1A
    .byte $02
    .byte $01
    .byte $00
    .byte $00

.segment "CODE"

.scope Scope
    zp_valid = $02

    .scope Inner
        zp_valid_too = $03
    .endscope
.endscope

zp_used    = $00                ; asan:reserve $02
whatever   = $01

lda zp_used
lda $00
lda whatever
lda Scope::zp_valid
lda Scope::Inner::zp_valid_too
lda zp_used + 1
lda zp_used + 2
lda zp_used - 1

@something:
    beq @something

ldx #0
lda palettes, x

palettes:
    .byte $0F
