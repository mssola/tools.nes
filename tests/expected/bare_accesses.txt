warning: accessing a memory region without using a variable (bare_accesses.s: line 22)
warning: accessing a memory region without a proper name ('whatever') (bare_accesses.s: line 23)
warning: accessing a memory region without using a variable (bare_accesses.s: line 40)
error: out of bounds memory access for 'zp_used' ($00-$01) (bare_accesses.s: line 27)
error: out of bounds memory access for 'zp_used' ($00-$01) (bare_accesses.s: line 28)
